core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java

```diff
@@ -327,11 +340,54 @@ public void seek(long pos) throws IOException {
     if (pos == mPos) {
       return;
     }
+    // Protect the original seek logic under fuse flag to minimize and isolate
+    // the fuse related changes
+    if (!mSharedCacheReader) {
+      if (pos < mPos) {
+        mEOF = false;
+      }
+      closeDataReader();
+      mPos = pos;
+      return;
+    }
+
     if (pos < mPos) {
       mEOF = false;
-    }
+      if (mDataReader instanceof SharedGrpcDataReader) {
+        SharedGrpcDataReader reader = (SharedGrpcDataReader) mDataReader;
+        reader.seek(pos);
+        if (mCurrentChunk != null) {
+          mCurrentChunk.release();
+          mCurrentChunk = null;
+        }
+      } else {
+        closeDataReader();
+      }
+    } else {
+      // TODO(lu) combine the original seek logic and the following general improvements
```

Comment:
This block looks like the seek logic for when we are NOT using the shared grpc reader. Is there a reason why we cannot just have 2 types of seek logic (1) using shared grpc reader, (2) without shared grpc reader?

Instead, we have 3 types of seek logic now.

Reply:
put the newly added two seek logic separately guarded by the flag.
in this way we introduce minimal change to existing codepath

Reply:
I changed the code a bit. so this code path is only executed when mDataReader instanceof SharedGrpcDataReader